diff --git a/contracts/Commitment/Commitment.sol b/contracts/Commitment/Commitment.sol
index df2bff9..7661c41 100644
--- a/contracts/Commitment/Commitment.sol
+++ b/contracts/Commitment/Commitment.sol
@@ -453,6 +453,8 @@ contract Commitment is
         // Whitelist part
         WhitelistTicket storage ticket = _whitelist[msg.sender];
         bool whitelisted = ticket.token == tokenType;
+        require(whitelisted || state() == State.Public);
+
         bool whitelistActiveForToken = tokenType == Token.Euro || state() == State.Whitelist;
         if (whitelisted && whitelistActiveForToken) {
             uint256 ticketEur = min(remainingEur, ticket.amountEur);
@@ -468,14 +470,10 @@ contract Commitment is
         }
 
         // Curve
-        if (whitelisted || state() == State.Public) {
+        if (remainingEur > 0) {
             totalNmk = add(totalNmk, NEUMARK.issueForEuro(remainingEur));
-            remainingEur = 0;
         }
 
-        // We don't do partial tickets
-        require(remainingEur == 0);
-
         // We don't go over the cap
         require(NEUMARK.totalEuroUlps() <= CAP_EUR);
 
