diff --git a/contracts/Commitment/Commitment.sol b/contracts/Commitment/Commitment.sol
index 0a7ca0e..63912a0 100644
--- a/contracts/Commitment/Commitment.sol
+++ b/contracts/Commitment/Commitment.sol
@@ -485,16 +485,26 @@ contract Commitment is
         uint256 amountEur = isEuro ? amount : convertToEur(amount);
         require(amount == 0 || amountEur >= MIN_TICKET_EUR);
 
-        // Allocate Neumarks (will be issued to `this`)
-        uint256 rewardNmk = NEUMARK.issueForEuro(amountEur);
+        // Register the investor on the list of investors to keep them
+        // in order.
+        _whitelistInvestors.push(investor);
 
-        // Add to pre-allocated tickets
+        // Create a ticket without NEUMARK reward information and add it to
+        // the pre-allocated tickets.
         _whitelist[investor] = WhitelistTicket({
             token: token,
             amountEur: amountEur,
-            rewardNmk: rewardNmk
+            rewardNmk: 0
         });
-        _whitelistInvestors.push(investor);
+
+        // Allocate Neumarks (will be issued to `this`).
+        // Because `_whitelist[investor].token == Token.None` does not not hold
+        // any more, this function is protected against reentrancy attack
+        // conducted from NEUMARK.issueForEuro().
+        uint256 rewardNmk = NEUMARK.issueForEuro(amountEur);
+
+        // Record the number of Neumarks for investor.
+        _whitelist[investor].rewardNmk = rewardNmk;
 
         // Add to totals
         // AUDIT[CHF-41] Use isEuro instead of isEther.
