diff --git a/contracts/Commitment/TimedStateMachine.sol b/contracts/Commitment/TimedStateMachine.sol
index 2ab41c0..a1ffa2f 100644
--- a/contracts/Commitment/TimedStateMachine.sol
+++ b/contracts/Commitment/TimedStateMachine.sol
@@ -67,19 +67,18 @@ contract TimedStateMachine is StateMachine {
     function handleTimedTransitions()
         public
     {
-        // Time relative to WHITELIST_START
-        int256 t = int256(block.timestamp) - WHITELIST_START;
+        int256 t = int256(block.timestamp);
 
         // Time induced state transitions.
         // @dev Don't use `else if` and keep sorted by time and call `state()`
         //     or else multiple transitions won't cascade properly.
-        if (state() == State.Before && t >= 0) {
+        if (state() == State.Before && t >= startOf(State.Whitelist)) {
             transitionTo(State.Whitelist);
         }
-        if (state() == State.Whitelist && t >= PUBLIC_FROM_START) {
+        if (state() == State.Whitelist && t >= startOf(State.Public)) {
             transitionTo(State.Public);
         }
-        if (state() == State.Public && t >= FINISH_FROM_START) {
+        if (state() == State.Public && t >= startOf(State.Finished)) {
             transitionTo(State.Finished);
         }
     }
